ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/healthcheck/HealthCheckAgent.java

```diff
@@ -84,7 +87,15 @@ public HealthCheckResponse checkHealth() {
         ));
     final boolean allHealthy = results.values().stream()
         .allMatch(HealthCheckResponseDetail::getIsHealthy);
-    return new HealthCheckResponse(allHealthy, results);
+    final Map<String, Set<TaskMetadata>> metadata = ksqlClient
+            .makeClusterStatusRequest(serverEndpoint)
```

Comment:
I am not sure if the correct thing todo was this, I could maybe make a new endpoint but that seemed even more disruptive. Also does this query the one instance or the whole cluster. If it is the later we can have the health check just hit this instead of adding it to the health check response.

Reply:
I am thinking if we should return the overall status of the queries in the whole cluster. If we need a more detailed response, maybe a new endpoint should be created (say `/diagnostics` or something) that returns the issues per instance?

Reply:
I'm not sure I follow. what do you mean by "the overall status of the queries in the whole cluster"?

The health check is hitting the `/healthcheck` endpoint. are you saying it should get the taskMetadata from a `/diagnostics` endpoint of the status request here should hit a more targeted endpoint than the entire `ClusterStatus`